### DISCUSSION TITLE
Change generated folder.

### DISCUSSION THREAD

**Discussion by Spriggan1992** [serverpod/serverpod#652] [Q&A] [answered]
Is there has way to change generated endpoints folder and protocols?
For example, I tried to change folder with endpoint but in client side cod wasn't generated.
In documentation, I didn't find any information about it. Maybe I miss something

**Comment by fischerscode** [comment]
As far as I know, the paths can currently not be changed. Might be a nice feature, though.

**Comment by vlidholt** [comment, marked as answer]
The endpoints need to go in the endpoints directory. You can specify where your files are places in the `config/generate.yaml` file.

**Reply by Spriggan1992** [reply]
Hi @vlidholt. Thank you to create for us such a cool server side app. Serverpod looks really awesome :)
May I ask you to provide an example, how exactly path should be specified in config/generated.yaml file? It would help me a lot.

**Comment by vlidholt** [comment]
This is the contents of the default `config/generator.yaml`:

```yaml
type: internal

client_package_path: ../serverpod_service_client
```

If you need to change the location of your client package, you can specify a new location there.

**Reply by Rockfella** [reply]
When I run the serverpod generate my files are moved into a new directory please see below, me and my colleauge are running the same versions of serverpod, and flutter, he is on mac and me windows. His command does not generate the "models" directory and mine does.

.yaml file location
server\lib\src\models\site.spy.yaml 
server\lib\src\models\customer.spy.yaml ----------------------------------------------------------------------------

Outcome: 
client/lib/src/protocol/customer.dart 
client/lib/src/protocol/models/customer.dart 
client/lib/src/protocol/site.dart 
client/lib/src/protocol/models/site.dart 

server/lib/src/generated/customer.dart 
server/lib/src/generated/models/customer.dart 
server/lib/src/generated/site.dart 
server/lib/src/generated/models/site.dart

Any idea how this could happen? Thanks

**Reply by Isakdl** [reply]
Can you confirm you both are running the same CLI version? the "models" directory was introduced in 1.2 and will not work with any version prior to 1.2.

`$ serverpod version`

**Reply by Rockfella** [reply]
@Isakdl We can confirm that we are running on the same CLI version, in addition my colleauge installed the same set-up on his windows pc and got the same issue like me. He tested 1.2.2. och 1.2.3. He tested the same on two mac´s without issues. 
We have tested it on brand new serverpod projects as well.. 

reproduce on windows:
serverpod create foo
cd foo_server
serverpod generate
git init .
git add -A
git commit -m "foo"
// modify example.spy.yaml, add field "foo: String"
serverpod generate

**Reply by Isakdl** [reply]
Yes I can confirm there is a bug on Windows, ill take a look at that asap. Thank you @Rockfella

**Reply by Isakdl** [reply]
@Rockfella fix is merged, next version (1.2.4) going out soon with the update.

**Reply by Rockfella** [reply]
Thats awesome, thank you for developing serverpod!
